horovod/spark/lightning/remote.py

```diff
@@ -106,22 +103,25 @@ def train(serialized_model):
             if train_logger is None:
                 train_logger = TensorBoardLogger(logs_path)
 
-            # TODO: find out a way to use ckpt_path created from remote store, but all other parameters ingest from estimator config
-            # ckpt_path = os.path.join(run_output_dir, remote_store.checkpoint_filename)
-            # os.makedirs(ckpt_path, exist_ok=True)
-            # model_checkpoint_callback = ModelCheckpoint(dirpath=ckpt_path)
-            # callbacks.append(model_checkpoint_callback)
-
-            is_model_checkpoint_callback_exist = False
-            for cb in callbacks:
-                if isinstance(cb, ModelCheckpoint):
-                    is_model_checkpoint_callback_exist = True
-                    break
+            # Lightning requires to add cehckpoint callbacks for all ranks.
+            # Otherwise we are seeing hanging in training.
+            _checkpoint_callback = checkpoint_callback
+            if _checkpoint_callback:
```

Comment:
_checkpoint_callback from user can specify what monitor to use? So that we will save other checkpoint than the last one?

Reply:
yes. user can customize its own by referring lightning documents https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.model_checkpoint.html